### PR TITLE
PUB-1772 Add functional test for Publication Management Service

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1772

### Change description ###

Add functional test for publication management service. INTERNAL_ERROR_CODE functional tests are missing because error will only occurred when there is an invalid json which is not possible. We cannot upload invalid json to the system.

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

**Review by joshblackmoor — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    @Test
+    @WithMockUser(username = "unknown_user", authorities = {"APPROLE_api.request.unknown"})
+    void testGenerateArtefactSummaryUnauthorized() throws Exception {
+        mockMvc.perform(get(GET_ARTEFACT_SUMMARY + "/" + ARTEFACT_ID))
+            .andExpect(status().isForbidden());
+    }
+
+    @Test
+    void testGenerateFileAccepted() throws Exception {
```
Comment (KianKwa) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java:95]: Think the original tests in pub services actually do a parameterized test for all list type. I wonder if that should be done here. Can discuss about that

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        file = new MockMultipartFile("file", "test.pdf",
+                                     MediaType.APPLICATION_PDF_VALUE, "test content".getBytes(
+            StandardCharsets.UTF_8));
+
+        objectMapper = new ObjectMapper();
+        objectMapper.findAndRegisterModules();
+    }
+
+    @Test
+    void testGenerateArtefactSummaryOK() throws Exception {
```
Comment (KianKwa) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java:68]: Same as the other comment. Need to decide whether we should add tests for all lists

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java]: As discussed, update the test.

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/channel/management/controllers/PublicationManagementTest.java]: As discussed, update the test.

**Review by KianKwa — APPROVED** [review]

**Pull request merged**